tests/main/snap-quota-groups/task.yaml

```diff
@@ -0,0 +1,76 @@
+  effective in practice.
+
+prepare: |
+  if os.query is-trusty; then
+    exit 0
+  fi
+  snap install go-example-webserver jq remarshal
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  if os.query is-trusty; then
+    # just check that we can't do anything with quota groups on trusty, systemd
+    # there is 204, but we need at least 205 for slice units
+
+    snap quota foobar --max-memory=1MB 2>&1 | MATCH "systemd version too old: snap quotas requires systemd 205 and newer \(currently have 204\)"
+    exit 0
+  fi
+
+  echo "Create a group with a snap in it"
+  snap quota group-one --max-memory=100MB go-example-webserver
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should also still be active"
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+active"
+
+  echo "The systemd slice should have one process in it now"
+  cgroupProcsFile="/sys/fs/cgroup/memory/$sliceName/snap.go-example-webserver.webserver.service/cgroup.procs"
+  wc -l < "$cgroupProcsFile" | MATCH 1
+  SERVER_PID=$(cat "$cgroupProcsFile")
+
+  echo "And that process is the main PID for the example webserver"
+  systemctl show --property=MainPID snap.go-example-webserver.webserver.service | MATCH "MainPID=$SERVER_PID"
+
+  echo "And the service is in the Control Group for the slice"
+  test "$(systemctl show --property=ControlGroup snap.go-example-webserver.webserver.service)" = "ControlGroup=/$sliceName/snap.go-example-webserver.webserver.service"
+
+  # TODO: check that systemctl says the memory usage is the same as "snap quota group-one"
+
+  echo "Removing the quota will stop the slice and the service will be restarted"
+  snap remove-quota group-one
+  systemctl show --property=MainPID snap.go-example-webserver.webserver.service | NOMATCH "MainPID=$SERVER_PID"
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+active"
+
+  echo "And the service is not in a slice anymore"
+  systemctl show --property=ControlGroup snap.go-example-webserver.webserver.service | NOMATCH "/$sliceName/snap.go-example-webserver.webserver.service"
+
+  echo "Creating a quota with a very small memory limit results in the service being unable to start"
+  snap quota too-small --max-memory=1MB go-example-webserver
+
+  echo "The systemd slice should be active"
+  sliceName="snap.$(systemd-escape --path too-small).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "But the service is not running after a short amount of time"
+  sleep 10 # meh probably unnecessarily long
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+inactive"
```

Comment:
I think it would be good to use retry-tool here with a short sleep time and a count of 10 or so, wasting 10 seconds every time is a bit of a waste.

Reply:
Yeah I didn't use it since I thought it didn't work with `sh -c ...` but actually it does since MATCH is not a bash function it's an actual program on $PATH so this works fine. Thanks for the suggestion